### PR TITLE
Integrate CloudBeaver

### PR DESCRIPTION
**Pull request by JDarDagran**

### CODE REVIEW

File: jupyterlab-dataops/Dockerfile [hunk 1]
```diff
@@ -16,12 +16,44 @@ COPY requirements.txt requirements-${PROVIDER}.tx[t] $HOME/
...
+    cd /tmp/cloudbeaver
+
+# Add BigQuery drivers
+RUN mkdir -p /tmp/cloudbeaver/server/drivers/bigquery/src/main/resources
+ADD resources/cloudbeaver/plugin.xml /tmp/cloudbeaver/server/bundles/io.cloudbeaver.resources.drivers.base/plugin.xml
+ADD resources/cloudbeaver/drivers/pom.xml /tmp/cloudbeaver/server/drivers/pom.xml
+ADD resources/cloudbeaver/drivers/bigquery/pom.xml /tmp/cloudbeaver/server/drivers/bigquery/pom.xml
+RUN wget https://storage.googleapis.com/simba-bq-release/jdbc/SimbaJDBCDriverforGoogleBigQuery42_1.2.16.1020.zip && \
+    unzip SimbaJDBCDriverforGoogleBigQuery42_1.2.16.1020.zip -d /tmp/cloudbeaver/server/drivers/bigquery/src/main/resources && \
+    rm SimbaJDBCDriverforGoogleBigQuery42_1.2.16.1020.zip
```
Comment (p-pekala) [inline jupyterlab-dataops/Dockerfile:53]: BigQuery / gcp can't be hardcoded this way. Look at github actions pipliens, platform is driven this way:     
strategy:
      matrix:
        provider: ["bigquery"]

File: jupyterlab-dataops/resources/cloudbeaver/conf/initial-data-sources.conf [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+	"connections": {
+		"google_bigquery_jdbc_simba-1": {
+			"provider": "bigquery",
+			"driver": "google_bigquery_jdbc_simba",
+			"name": "Google BigQuery",
+			"save-password": true,
+			"show-system-objects": true,
+			"read-only": false,
+			"configuration": {
+				"host": "https://www.googleapis.com/bigquery/v2",
```
Comment (p-pekala) [inline jupyterlab-dataops/resources/cloudbeaver/conf/initial-data-sources.conf:12]: same here, we are hardcoding GCP, do we want it?

File: jupyterlab-dataops/resources/cloudbeaver/conf/initial-data-sources.conf [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+				"port": "443",
+				"url": "jdbc:bigquery://https://www.googleapis.com/bigquery/v2:443",
+				"type": "dev",
+				"properties": {
+					"OAuthPvtKeyPath": "/var/gcp-service-account-key.json"
```
Comment (p-pekala) [inline jupyterlab-dataops/resources/cloudbeaver/conf/initial-data-sources.conf:17]: Is it uploaded manually by a user? Meybe home dir is better place?

File: jupyterlab-dataops/resources/cloudbeaver/conf/initial-data.conf [hunk 4]
```diff
@@ -0,0 +1,16 @@
+{
+    roles: [
+        {
+            roleId: "admin",
+            name: "Admin",
+            description: "Administrative access. Has all permissions.",
+            permissions: [ "public", "admin" ]
+        },
```
Comment (p-pekala) [inline jupyterlab-dataops/resources/cloudbeaver/conf/initial-data.conf:8]: These are roles for dbever itself?

File: jupyterlab-dataops/resources/cloudbeaver/drivers/pom.xml [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+    <modules>
+        <module>bigquery</module>
+        <module>h2</module>
+        <module>mysql</module>
+        <module>mariadb</module>
+        <module>oracle</module>
+        <module>postgresql</module>
+        <module>sqlserver</module>
+        <module>trino</module>
```
Comment (p-pekala) [inline jupyterlab-dataops/resources/cloudbeaver/drivers/pom.xml:21]: are all these modules used?

Comment (JDarDagran) [reply jupyterlab-dataops/resources/cloudbeaver/drivers/pom.xml:21]: That's default config from cloudbeaver.  Only BigQuery is added here. All the others are enabled but we can filter which should be used.

File: jupyterlab-dataops/resources/cloudbeaver/conf/initial-data-sources.conf (hunk 2 above)
Comment (JDarDagran) [reply jupyterlab-dataops/resources/cloudbeaver/conf/initial-data-sources.conf:12]: this endpoint stays the same AFAIK for bigquery connection

Comment (JDarDagran) [reply jupyterlab-dataops/resources/cloudbeaver/conf/initial-data-sources.conf:12]: It serves only as template connection though. The connection itself should be configured in some external by uploading/setting service account JSON. We could get rid of this configuration here, no?

File: jupyterlab-dataops/Dockerfile [hunk 6]
```diff
@@ -16,12 +16,44 @@ COPY requirements.txt requirements-${PROVIDER}.tx[t] $HOME/
...
+# Clone cloudbeaver
+RUN git clone https://github.com/dbeaver/cloudbeaver.git /tmp/cloudbeaver && \
+    cd /tmp/cloudbeaver
+
+# Add BigQuery drivers
+RUN if [[ $PROVIDER == "bigquery" ]] ; then mkdir -p /tmp/cloudbeaver/server/drivers/bigquery/src/main/resources ; fi
+ADD resources/cloudbeaver/plugin-${PROVIDER:-default}.xm[l] /tmp/cloudbeaver/server/bundles/io.cloudbeaver.resources.drivers.base/plugin.xml
+ADD resources/cloudbeaver/drivers/pom-${PROVIDER:-default}.xm[l] /tmp/cloudbeaver/server/drivers/pom.xml
+ADD resources/cloudbeaver/drivers/${PROVIDER}/pom.xm[l] /tmp/cloudbeaver/server/drivers/bigquery/pom.xml
+RUN if [[ $PROVIDER == "bigquery" ]] ; then wget https://storage.googleapis.com/simba-bq-release/jdbc/SimbaJDBCDriverforGoogleBigQuery42_1.2.16.1020.zip && \
```
Comment (p-pekala) [inline jupyterlab-dataops/Dockerfile:51]: can we merge two sesions with provider == bigquery?

**Review by p-pekala — APPROVED** [review]

File: jupyterlab-dataops/Dockerfile (hunk 6 above)
Comment (JDarDagran) [reply jupyterlab-dataops/Dockerfile]: changed

**Pull request merged**